### PR TITLE
lxd: network.limit-related bugfixes

### PR DESCRIPTION
**Pull request by mihalicyn**
https://github.com/canonical/lxd/pull/12135#issuecomment-1687736767

### CODE REVIEW

File: lxd/instance/drivers/driver_lxc.go [hunk 1]
```diff
@@ -2564,6 +2564,19 @@ func (d *lxc) onStart(_ map[string]string) error {
 	if d.expandedConfig["limits.network.priority"] != "" {
 		go func(d *lxc) {
 			d.fromHook = false
+
+			timeout := time.Now().Add(10 * time.Second)
```
Comment (tomponline) [inline lxd/instance/drivers/driver_lxc.go:2568]: @mihalicyn can you check whether we can avoid needing this loop by instead changing the check in `setNetworkPriority` to use `if c.InitPID() <= 0 {` instead, like we do in `deviceTaskBalance`?

**Comment by tomponline** [comment]
@mihalicyn did you rebase your branch on main btw as the doc builds are failing?

File: lxd/instance/drivers/driver_lxc.go (hunk 1 above)
Comment (mihalicyn) [reply lxd/instance/drivers/driver_lxc.go]: great idea! Have done.

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
Thanks!

**Pull request merged**